qemu/tests/vsock_hotplug.py

```diff
@@ -0,0 +1,71 @@
+import re
+import logging
+
+from avocado.utils import linux_modules
+
+from virttest import error_context
+from virttest import utils_vsock
+from virttest.qemu_devices import qdevices
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Hotplug/unhotplug virtio-vsock device
+
+    1. Boot guest without virtio-vsock-pci device
+    2. Hotplug virtio-vsock device
+    3. Check device from qemu level(info pci)
+    4. Check device inside guest(lspci/dmesg)
+    5. Unplug virtio-vsock device
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment
+    """
+
+    linux_modules.load_module('vhost_vsock')
+    vm = env.get_vm(params['main_vm'])
+    session = vm.wait_for_login()
+    guest_cid = utils_vsock.get_guest_cid(3)
+    vsock_id = 'hotplugged_vsock'
+    vsock_params = {'id': vsock_id, 'guest-cid': guest_cid}
+    if '-mmio:' in params.get('machine_type'):
+        dev_vsock = qdevices.QDevice('vhost-vsock-device', vsock_params)
+    elif params.get('machine_type').startswith("s390"):
+        dev_vsock = qdevices.QDevice("vhost-vsock-ccw", vsock_params)
+    else:
+        dev_vsock = qdevices.QDevice('vhost-vsock-pci', vsock_params)
+    vm.devices.simple_hotplug(dev_vsock, vm.monitor)
+
+    error_msg = ''
+    error_context.context('Check device %s exist in qemu monior command'
+                          ' lspci output.' % vsock_id, logging.info)
+    monitor_pci_info = vm.monitor.info('pci')
+    if not re.search(vsock_id, str(monitor_pci_info)):
+        error_msg += 'monitor pci check failed.\\n'
```

Comment:
In `vm.devices.simple_hotplug`, it will check the device id with "info qtree", do we still need to check the id with 'info pci' ?

Reply:
@yanan-fu That's for matching the test case precisely, @xiangchunfu please help confirm the steps and check pionts. Thanks.

Reply:
@vivianQizhu @yanan-fu  I double-checked this issue with info qtree.  If vm.device.simple_hotplug can check vsock device id like this:    It is enough for this case.   purpose is:  make sure can find this device from qemu level layer. 
      type PCI
      dev: vhost-vsock-pci, id "vhost-vsock-pci0"

Reply:
@xiangchunfu @yanan-fu Thanks, I will remove the `info pci` part.

Reply:
@yanan-fu  if the first step failed,  don't need to check other steps.